src/Stateless/StateConfiguration.cs

```diff
@@ -508,6 +508,26 @@ public StateConfiguration OnExit(Action<Transition> exitAction, string exitActio
             /// <returns>The receiver.</returns>
             public StateConfiguration SubstateOf(TState superstate)
             {
+                var State = _representation.UnderlyingState;
+
+                // Check for accidental identical cyclic configuration 
+                if (State.Equals(superstate)) { throw new ArgumentException($"Configuring {State} as a substate of {superstate} creates an illegal cyclic configuration."); }
```

Comment:
Nit: apart from argument null checks, `if` statements in the repo are mostly over multiple lines. Braces optional, but if they're present they should be on a line each.